gremlin-dotnet/src/Gremlin.Net/Process/Traversal/GraphTraversal.cs

```diff
@@ -96,15 +99,15 @@ private GraphTraversal(ICollection<ITraversalStrategy> traversalStrategies, Byte
         /// <summary>
         ///     Adds the E step to this <see cref="GraphTraversal{SType, EType}" />.
         /// </summary>
-        public GraphTraversal<TStart, Edge> E (params object[] edgeIdsOrElements)
+        public GraphTraversal<TStart, Edge> E (params object?[]? edgeIdsOrElements)
         {
             if (edgeIdsOrElements == null)
             {
-                Bytecode.AddStep("E", new object[] { null });
+                Bytecode.AddStep("E", new object?[] { null });
             }
             else
             {
-                var args = new List<object>(edgeIdsOrElements.Length);
+                var args = new List<object?>(edgeIdsOrElements.Length);
```

Comment:
why is the `edgeIdsOrElements` array copied through the list?

Reply:
This file was originally generated automatically by some Groovy script that created basically one C# implementation of every Java implementation of a Gremlin step. It simply created a new list for all arguments so it could add all arguments to that list and then pass the list to `Bytecode.AddStep()`.
Now that we no longer generate this code automatically, we could of course improve / optimize it by hand and remove these unnecessary creation of lists for example. However I think that should be done in a separate PR.